r-reproducibility.Rmd

```diff
@@ -0,0 +1,1211 @@
+2. Create a new R Markdown file, with output type "HTML", and save it as
+`mimic-html-file.Rmd`.
+3. Delete all the automatically added text except the top part Inside the R Markdown file
+4. Write text using Markdown formatting so that you
+can create a `html_document` that looks exactly like that (linked) file.
+4. Knit the 
+
+### R Markdown
+
+[R Markdown][r-markdown] is an extension of Markdown that weaves together R code
+with Markdown formatted text all together in a single document. Output from the
+R code gets inserted directly into the document for a seamless integration of
+document writing and analysis reporting.
+
+#### YAML header/metadata
+
+Most Markdown documents (especially for R Markdown) include [YAML][yaml] metadata at
+the top of the document, surrounded by `---`. The YAML contains the metadata and
+various options for the entire document. For instance, the title and/or author 
+text is included here, as well as the options on what the converted output
+document should be, such as Word, HTML, or PDF (if you want to create a beautiful
+PDF, you need to install the R package [tinytex][tinytex]). While these three
+are the most commonly used output formats, there are many more output formats to
+choose from (e.g. slides, websites, books). The YAML header looks something like
+this:
+
+```yaml
+---
+title: "Document title"
+author: Your Name
+output: html_document
+---
+```
+
+Usually when you create the R Markdown file, this YAML header gets added
+automatically. There are additional options you can set in the output field,
+which we will cover later on. In this particular case, the R Markdown file
+will convert to a HTML document via the `html_document` setting. There are also
+`word_document` and `pdf_document` settings. For now, we'll focus on the
+`html_document` output.
+
+#### Using R code chunks 
+
+R Markdown's primary function is to allow combining text and R code in the same
+document, which is incredibly powerful and useful! All R code chunks have the 
+appearance:
+
+````markdown
+`r ''````{r chunk-label-name, chunk.option="...", chunk.option=...}
+...R code...
+`r ''````
+````
+
+Notice that chunk options need to be on the same line (one single line). Any
+R code in the code chunk gets evaluated and any output gets inserted into the
+document. So if the code prints to the console, it will get inserted into the 
+document.
+
+The `r` tells R Markdown to run R on the code chunk, while the [chunk label][chunk-label]
+differentiates the chunk from other chunks. Using a chunk label also helps navigate
+a document when you use RStudio's "Document outline" (`Code -> Show Document Outline`).
+It also does a few other things.
+
+**Note**: Code [chunk labels][chunk-label] should be named without `_`, spaces,
+or `.` and instead should be one word or be separated by `-`. While an error may
+not necessarily occur, there can be some unintended side effects that will cause
+you some annoyance without knowing the reason.
```

Comment:
Such as? (If an error is possible, please show learners how to tell if that's what they've tripped over.)

Reply:
The errors only occur in really specialized settings which I personally don't encounter almost ever. I only encountered it once and I think it was related to building websites. I think it might be too much of a detour, so I'll simply say its "standard practice".